File: 04-recursive-sum/recursive_sum.cpp
#include <iostream>
#include <fstream>
using namespace std;

int recursiveSum(int n) {
    // Base case
    if (n == 1) {
        return 1;
    }
    
    // Recursive case
    return n + recursiveSum(n - 1);
}

int main() {
    // Read input
    ifstream input("input.txt");
    int n;
    input >> n;
    input.close();
    
    // Calculate result
    int result = recursiveSum(n);
    
    // Write output
    ofstream outFile("output.txt");
    outFile << result << endl;
    
    return 0;
} 

File: 04-recursive-sum/recursive_sum.py
def recursive_sum(n: int) -> int:
    # Base case
    if n == 1:
        return 1
    
    # Recursive case
    return n + recursive_sum(n - 1)

def main():
    # Read input
    with open('input.txt', 'r') as f:
        n = int(f.readline())
    
    # Calculate result
    result = recursive_sum(n)
    
    # Write output
    with open('output.txt', 'w') as f:
        f.write(str(result) + '\n')

if __name__ == "__main__":
    main() 